source/posix/thread.c

```diff
@@ -460,3 +460,16 @@ int aws_thread_current_at_exit(aws_thread_atexit_fn *callback, void *user_data)
     tl_wrapper->atexit = cb;
     return AWS_OP_SUCCESS;
 }
+
+struct aws_string *aws_thread_current_name(struct aws_allocator *allocator) {
+#ifdef AWS_PTHREAD_HAS_GETNAME
+    char name[16] = {0};
+    if (pthread_getname_np(aws_thread_current_thread_id(), name, 16)) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
```

Comment:
there's no way OUR user could have passed an invalid arg to this function, so this error isn't helping them

I'd have AWS_ERROR_SYS_CALL_FAILURE be "we don't know what went wrong" error